src/Libraries/RevitServices/Persistence/ElementBinder.cs

```diff
@@ -47,6 +47,34 @@ public SerializableId(SerializationInfo info, StreamingContext context)
             StringID = (string) info.GetValue("stringID", typeof (string));
             IntID = (int)info.GetValue("intID", typeof(int));
         }
+
+        public override bool Equals(object other)
+        {
+            var sID = other as SerializableId;
+            if (sID == null)
+            {
+                return false;
+            }
+
+            if (this.IntID.Equals(sID.IntID) && this.StringID.Equals(sID.StringID))
+            {
+                return true;
+            }
+            return false;
+
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked // Overflow is fine, just wrap
+            {
+                int hash = 17;
+
+                hash = hash * 23 + StringID.GetHashCode();
+                hash = hash * 23 + IntID.ToString().GetHashCode();
+                return hash;
```

Comment:
I'm not too experienced with hash code generation, so I'd typically do this (not saying yours is wrong, just wondering how that is derived):

```csharp
return (StringID.GetHashCode() ^ IntID.GetHashCode());
```